### PR TITLE
Document how and when the library raises exceptions

### PR DESCRIPTION
**Pull request by timrogers**
This adds a section to the readme which explains when the library will raise exceptions and how you can work with them.

I did some work with `octokit/octokit.rb` and `octokit/octokit.js` recently, and I found that not having this meant that I had to do a bunch of digging myself to understand how I should handle unhappy paths.

As well as adding the error handling documentation, this PR also makes a few other small changes to the README: correcting, renaming and restructuring headings.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,6 +148,23 @@ response  = client.last_response
...
 
+## Handling errors
+
+When the API returns an error response, Octokit will raise a Ruby exception.
+
+A range of different exceptions can be raised depending on the error returned
+by the API - for example:
+
+* A `400 Bad Request` response will lead to an `Octokit::BadRequest` error 
+* a `403 Forbidden` error with a "rate limited exceeded" message will lead
```
Comment (nickfloyd) [inline README.md:159]: Just a small nit.

```suggestion
* A `403 Forbidden` error with a "rate limited exceeded" message will lead
```

**Review by nickfloyd — CHANGES_REQUESTED** [review]
Hey @timrogers thanks for these additions!  I just left one small nit on the bulleted items for the errors.  Let me know what you think!

**Review by nickfloyd — APPROVED** [review]
👍  LGTM!

**Comment by hmharvey** [comment]
> 👍 LGTM! Side note - did you intend your base branch to be octokit:4-stable - This feels like the base should be master given it's just a docs change and not a hotfix.

`octokit:4-stable` is still the correct branch! `master` has significantly diverged at this point 😄

**Comment by timrogers** [comment]
I'm not allowed to merge this, which I'm guessing is due to the outstanding GitHub checks (which aren't needed due to the `ci skip`!). 

I'll just let the checks run for the last commit to get around this 🙏

**Comment by timrogers** [comment]
> `octokit:4-stable` is still the correct branch! `master` has significantly diverged at this point 😄 

Interesting! 🤔 Is there some clean-up we can or should do here? Perhaps we could delete `master` and create a new `main` branch starting from `4-stable`?

**Comment by nickfloyd** [comment]
> Interesting! 🤔 Is there some clean-up we can or should do here? Perhaps we could delete `master` and create a new `main` branch starting from `4-stable`?

Yeah, we've got plans to get things cinched up around builds, workflow, and CI. Great work has already been done on octokit Ruby; it will take a bit of effort, but I am definitely looking forward to tackling things like these so that we can get this to a point where it is up to date and more welcoming to contributors!

**Comment by timrogers** [comment]
> > Interesting! 🤔 Is there some clean-up we can or should do here? Perhaps we could delete `master` and create a new `main` branch starting from `4-stable`?
> 
> Yeah, we've got plans to get things cinched up around builds, workflow, and CI. Great work has already been done on octokit Ruby; it will take a bit of effort, but I am definitely looking forward to tackling things like these so that we can get this to a point where it is up to date and more welcoming to contributors!

Wonderful! 🙌🏻

**Comment by timrogers** [comment]
@nickfloyd This should be ready for merge now. Thank you so much!

**Comment by imwiss** [comment]
Thanks for updating this, @timrogers! ✨

**Pull request merged**